lighthouse-core/audits/byte-efficiency/legacy-javascript.js

```diff
@@ -372,11 +372,8 @@ class LegacyJavascript extends ByteEfficiencyAudit {
     let transferRatio = transferRatioByUrl.get(url);
     if (transferRatio !== undefined) return transferRatio;
 
-    const mainDocumentRecord = NetworkAnalyzer.findOptionalMainDocument(networkRecords);
```

Comment:
Using `findOptionalMainDocument` without specifying a URL will just find the first document request. This isn't necessarily the main document if there was a redirect though.

I think it makes more sense to just get the main document record using the script URL like any other script in this case. For timespan/snapshot, `finalUrl` isn't necessarily the main document URL either https://github.com/GoogleChrome/lighthouse/issues/13706.

Reply:
yup! and this would very much be a bug for inline scripts in iframes. not sure why we (Read: I) did it this way 🤔